unittests/allplatformstests.py

```diff
@@ -3863,3 +3864,176 @@ def do_install(tags=None):
         self.assertEqual(sorted(expected_custom), do_install('custom'))
         self.assertEqual(sorted(expected_runtime_custom), do_install('runtime,custom'))
         self.assertEqual(sorted(expected_all), do_install())
+
+    def test_introspect_install_plan(self):
+        testdir = os.path.join(self.unit_test_dir, '99 install_plan')
+        introfile = os.path.join(self.builddir, 'meson-info', 'intro-install_plan.json')
+        self.init(testdir)
+        self.assertPathExists(introfile)
+        with open(introfile, encoding='utf-8') as fp:
+            res = json.load(fp)
+
+        env = get_fake_env(testdir, self.builddir, self.prefix)
+        cc = detect_c_compiler(env, MachineChoice.HOST)
+
+        def shared_lib_name(name):
```

Comment:
This is duplicated across:
- mesonbuild.build.SharedLibrary -> determine_filenames()
- run_project_tests.py -> InstalledFile
- this new third implementation

I'm not sure how easy, if at all, it will be to extract usable information out of the SharedLibrary class. :( However, the InstalledFile class is much more self-contained and doesn't have build state tied to it, so it should be possible to reuse here. Perhaps after some refactoring to extract a common "get_filename" method.

For bonus points get all 3 to use the same implementation. :D

Reply:
I re-used `mesonbuild.build.{Executable,SharedLibrary,StaticLibrary}`.